src/services/upower/device.hpp

```diff
@@ -117,6 +117,10 @@ class UPowerDevice: public QObject {
 	///
 	/// This will be equivalent to `type == Battery && powerSupply == true`.
  	Q_PROPERTY(bool isLaptopBattery READ isLaptopBattery NOTIFY isLaptopBatteryChanged);
+	/// Native path of the device specific to your OS.
+	Q_PROPERTY(QString nativePath READ nativePath NOTIFY nativePathChanged);
+	/// If the device is currently running on battery power, or discharging.
+	Q_PROPERTY(bool onBattery READ onBattery NOTIFY onBatteryChanged);
```

Comment:
OnBattery is a global property, not a device property, see https://upower.freedesktop.org/docs/UPower.html

Reply:
I apologize, that completely slipped by me!